installer/scripts/tomlparser.rb

```diff
@@ -117,19 +119,36 @@ def populateSettingValuesFromConfigMap(parsedConfig)
       puts "config::error::Exception while reading config settings for cluster level environment variable collection - #{errorStr}, using defaults"
     end
   end
+
+  begin
+    if !parsedConfig.nil?  && !parsedConfig[:agent_settings][:health_model].nil? && !parsedConfig[:agent_settings][:health_model][:enabled].nil?
+        @enable_health_model = parsedConfig[:agent_settings][:health_model][:enabled]
+        puts "enable_health_model = #{@enable_health_model}"
+    end
+  rescue => errorStr
+    puts "config::error:Exception while reading config settings for health_model enabled setting - #{errorStr}, using defaults"
+    @enable_health_model = false
+  end
 end
 
 @configSchemaVersion = ENV["AZMON_AGENT_CFG_SCHEMA_VERSION"]
 puts "****************Start Config Processing********************"
 if !@configSchemaVersion.nil? && !@configSchemaVersion.empty? && @configSchemaVersion.strip.casecmp("v1") == 0 #note v1 is the only supported schema version , so hardcoding it
-  configMapSettings = parseConfigMap
+    configMapSettings = {}
+
+    #iterate over every *settings file and build a hash of settings
+    Dir["/etc/config/settings/*settings"].each{|file|
```

Comment:
Won't this parse prometheus settings as well?

Reply:
It will, but they wont be processed, since we look for specific keys in populate settings value from config map